pallets/automation-time/src/lib.rs

```diff
@@ -1065,23 +1272,36 @@ pub mod pallet {
 				return Ok(task_id)
 			}
 
-			for time in execution_times.iter() {
-				match Self::get_scheduled_tasks(*time) {
-					None => {
-						let task_ids: BoundedVec<T::Hash, T::MaxTasksPerSlot> =
-							vec![task_id].try_into().unwrap();
-						<ScheduledTasks<T>>::insert(*time, task_ids);
-					},
-					Some(mut task_ids) => {
-						if let Err(_) = task_ids.try_push(task_id) {
-							return Err(Error::<T>::TimeSlotFull)?
-						}
-						<ScheduledTasks<T>>::insert(*time, task_ids);
-					},
+			Self::insert_scheduled_tasks(task_id, execution_times)
+		}
+
+		/// Insert task id into scheduled tasks
+		/// With transaction will protect against a partial success where N of M execution times might be full,
+		/// rolling back any successful insertions into the schedule task table.
+		fn insert_scheduled_tasks(
+			task_id: T::Hash,
+			execution_times: Vec<UnixTime>,
+		) -> Result<T::Hash, Error<T>> {
+			with_transaction(|| -> storage::TransactionOutcome<Result<T::Hash, DispatchError>> {
```

Comment:
@lreesby flagging this for your review in particular since you removed `with_transaction` in one of your pr. While trying to resolve the merge conflict I ended up putting it back in, because I'm calling this same method outside of the extrinsic context where I only want to revert a small part of the transaction.

Reply:
this line would be redundant then and can be removed for each extrinsic call.  https://github.com/OAK-Foundation/OAK-blockchain/pull/180/files#diff-e6a13f7032a383e19b77d20043ca46041d94f5e49396812e2de1b2000c301e59R601